src/Microsoft.AspNet.OData.Shared/EdmChangedObjectCollection.cs

```diff
@@ -57,5 +69,288 @@ private void Initialize(IEdmEntityType entityType)
             _edmType = new EdmDeltaCollectionType(new EdmEntityTypeReference(_entityType, isNullable: true));
             _edmTypeReference = new EdmCollectionTypeReference(_edmType);
         }
+
+        /// <summary>
+        /// Patch for Types without underlying CLR types
+        /// </summary>
+        /// <param name="originalCollection"></param>
+        /// <returns>ChangedObjectCollection response</returns>
+        public EdmChangedObjectCollection Patch(ICollection<EdmStructuredObject> originalCollection)
+        {            
+            PatchHandler = new DefaultTypelessPatchHandler(originalCollection, _entityType);
+
+            return CopyChangedValues();
+        }
+
+        /// <summary>
+        /// Handler for users Create, Get and Delete Methods
+        /// </summary>
+        internal TypelessPatchMethodHandler PatchHandler { get; set; }
```

Comment:
See related comments on DataSet<T>.  I would rather pass PatchHandler ot the Patch/CopyChangedValues method than have it an internal property on the EdmChangedObjectCollection.